### ISSUE TITLE
virtualbox guest additions grsecurity desktop

### ISSUE THREAD

**Issue by NikolaMandic** [NixOS/nixpkgs#16016]
## Issue description
it .12 version seems not to work with newer kernels and .20 version does not work properly 

did anybody try grsecurity with desktop package
 it does not seem to work with none except xfce
and then clipboard wont work
in the virtualbox I must say  no cp between host and guest
 anybody using it ?
gnome3 crashes kde4 crashes
and kde5 loads but nothing is shown
not sure if I had to do more than enable it
### Steps to reproduce

enable grsec and desktop in virtualbox with guest additions on latest nix

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
* Nix version: (run `nix-env --version`)
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

**Comment by joachifm** [comment]
I'm unclear on what you've done here: are you running virtualbox on a grsec kernel or grsec within virtualbox? Anyway, I'm still waiting on darwin builds to complete to staging can be merged into master. Until then, I'm inclined to say grsec on anything but a headless server is unlikely to be a good experience.

**Comment by NikolaMandic** [comment]
I am trying to run virtual machine that has nixos with grsecurity and desktop like kde4
so hardened desktop inside of a virtual machine

**Comment by joachifm** [comment]
I see, then this is a duplicate of https://github.com/NixOS/nixpkgs/issues/6905

**Comment by joachifm** [comment]
You'll probably need to compile your own kernel, using the grsec autoconfiguration method, with the following options enabled:
```
GRKERNSEC_CONFIG_AUTO
GRKERNSEC_CONFIG_DESKTOP
GRKERNSEC_CONFIG_VIRT_GUEST
GRKERNSEC_CONFIG_VIRT_VIRTUALBOX
```
I don't use virtualbox so it's not a priority for me.

**Comment by grahamc** [comment]
Is this enough of a dupe to close this one?
